client/settings/digital-wallets/test/index.test.js

```diff
@@ -30,6 +81,27 @@ describe( 'DigitalWallets', () => {
 		expect( cartCheckbox ).toBeDisabled();
 		expect( cartCheckbox ).not.toBeChecked();
 
+		expect( updateIsDigitalWalletsEnabledHandler ).toBeCalledTimes( 1 );
+	} );
+
+	it( 'should enable 1-click checkout sections if 1-click checkout is enabled', async () => {
+		const updateIsDigitalWalletsEnabledHandler = jest.fn();
+		useDigitalWalletsSettings.mockReturnValue(
+			getMockDigitalWalletsSettings(
+				true,
+				updateIsDigitalWalletsEnabledHandler
+			)
+		);
+
+		render( <DigitalWallets /> );
+
+		const [
+			,
+			checkoutCheckbox,
+			productPageCheckbox,
+			cartCheckbox,
+		] = screen.getAllByRole( 'checkbox' );
+
 		userEvent.click( screen.getByText( 'Enable 1-click checkouts' ) );
```

Comment:
Looks like this is no longer needed, because on this line the method is already enabled: https://github.com/Automattic/woocommerce-payments/blob/add/1728-save-digital-wallets-settings/client/settings/digital-wallets/test/index.test.js#L91
The "click" action is not updating any values.

Reply:
No, this is still needed. 

The "click" event is needed to trigger a state from "checked" to "unchecked". The title of this test case, however, needs to be updated to "should disable 1-click checkout sections if 1-click checkout is disabled"

Reply:
But on L91, the `getMockDigitalWalletsSettings` is called with `true` as the first argument. So in the tests, the `Enable 1-click checkouts` checkbox is already rendered in the "checked" state.
Firing an event here is not triggering any change on status, it's just triggering the `updateIsDigitalWalletsEnabledHandler` to be called.

Reply:
Correct, on line 91, the "Enable 1-click checkouts" is rendered as checked. Calling "click" should call the `updateIsDigitalWalletsEnabledHandler`. I want to make sure `updateIsDigitalWalletsEnabledHandler` is called when the checkbox is clicked. Line 84 verifies this call.

Am I missing something?

Reply:
The previous implementation used to actively change the value of the checkbox. But in this case the checkbox is instantiated with the checked/unchecked status on mount, due to the mocked values. So the "click" action is not performing any UI updates, it's just calling the handler function (`updateIsDigitalWalletsEnabledHandler`).

My "concern" is that if we were to refactor this functionality of the test by actually changing the internal state, the "click" action _would_ trigger a re-render and actively un-check the box, making the test fail.
I think the test could be better future-proofed by
- Removing the `userEvent.click` action
- Removing the expectation on `expect( updateIsDigitalWalletsEnabledHandler ).toBeCalledTimes( 1 )`
- Adding a new test that _just_ tests that on click of the checkbox, the `updateIsDigitalWalletsEnabledHandler` is being called.

I think that that situation would be avoid a possible issue with the test failing with a false negative in case of refactor.
Let me know if that makes sense!